caas/kubernetes/provider/pod.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package provider
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/core/status"
+
+	core "k8s.io/api/core/v1"
+)
+
+type EventGetter func() ([]core.Event, error)
+
+const (
+	ReasonCompleted                = "Completed"
+	ReasonContainersNotInitialized = "ContainersNotInitialized"
+	ReasonContainersNotReady       = "ContainersNotReady"
+	ReasonCrashLoopBackoff         = "CrashLoopBackOff"
+	ReasonError                    = "Error"
+	ReasonImagePull                = "ErrImagePull"
+)
+
+var (
+	podContainersReadyReasonsMap = map[string]status.Status{
+		ReasonContainersNotReady: status.Maintenance,
+	}
+
+	podInitializedReasonsMap = map[string]status.Status{
+		ReasonContainersNotInitialized: status.Maintenance,
+	}
+
+	podReadyReasonMap = map[string]status.Status{
+		ReasonContainersNotReady:       status.Maintenance,
+		ReasonContainersNotInitialized: status.Maintenance,
+	}
+
+	podScheduledReasonsMap = map[string]status.Status{
+		core.PodReasonUnschedulable: status.Blocked,
+	}
+)
+
+// GetPodCondition extracts the provided condition from the given status and returns that.
+// Returns nil and -1 if the condition is not present, and the index of the located condition.
+func GetPodCondition(status *core.PodStatus, conditionType core.PodConditionType) (int, *core.PodCondition) {
```

Comment:
I can't see that the index return result is ever used, do we need it?

Reply:
It was a straight copy and paste from the Kubernetes code base as we can't import these functions.

Reply:
Done & Done